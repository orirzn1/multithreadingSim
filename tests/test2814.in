L8
S7
O6
N8

T0
I@0x00000000
ADDI $4, $5, 0x1
ADDI $4, $7, 0x2
ADDI $4, $3, 0x2
SUB $4, $2, $7
ADD $5, $1, $3
SUBI $1, $7, 0x3
HALT $0

T1
I@0x00000000
SUB $4, $6, $3
STORE $1, $2, $6
ADDI $1, $4, 0x3
SUBI $3, $0, 0x4
LOAD $0, $2, 0x1
ADDI $3, $1, 0x1
SUBI $2, $6, 0x3
SUB $2, $6, $6
SUBI $2, $0, 0x2
HALT $0

T2
I@0x00000000
LOAD $2, $6, $3
SUBI $0, $5, 0x1
ADD $6, $2, $5
STORE $6, $7, 0x2
ADDI $4, $4, 0x3
LOAD $4, $5, $4
HALT $0

T3
I@0x00000000
ADD $2, $7, $0
LOAD $6, $1, $6
STORE $0, $1, 0x1
ADD $2, $2, $3
ADD $6, $7, $0
ADDI $5, $3, 0x4
SUB $0, $2, $0
HALT $0

T4
I@0x00000000
ADD $5, $6, $2
ADD $4, $5, $2
LOAD $2, $1, 0x1
ADD $6, $6, $3
SUB $6, $6, $2
STORE $0, $1, 0x2
ADDI $7, $6, 0x1
SUB $6, $6, $7
HALT $0

T5
I@0x00000000
ADDI $2, $1, 0x4
SUB $7, $0, $2
SUBI $7, $1, 0x2
SUB $6, $4, $3
STORE $6, $4, 0x1
ADD $5, $7, $2
ADD $1, $7, $0
LOAD $7, $5, 0x4
STORE $6, $7, $5
HALT $0

T6
I@0x00000000
SUB $3, $0, $0
SUBI $3, $7, 0x2
SUBI $4, $1, 0x1
ADD $0, $1, $2
STORE $4, $4, 0x1
SUBI $7, $4, 0x1
ADD $2, $6, $6
ADDI $3, $4, 0x1
HALT $0

T7
I@0x00000000
LOAD $1, $0, $7
ADDI $7, $2, 0x2
ADDI $3, $5, 0x1
ADD $4, $4, $0
ADDI $0, $2, 0x1
HALT $0

D@0x0
0x1
0x3
0x1
0x1
0x5
0xA
0x5
0xA
0x3
0x2
