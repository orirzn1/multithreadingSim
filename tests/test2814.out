
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x0	R1 = 0xFFFFFFFD	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0xFFFFFFFF	R1 = 0x3	R2 = 0xFFFFFFFD	R3 = 0x4	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0xFFFFFFFF	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x4	R6 = 0x0	R7 = 0x0
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFF	R7 = 0x1
Register file thread id 5:
	R0 = 0x0	R1 = 0xFFFFFFFE	R2 = 0x4	R3 = 0x0	R4 = 0x0	R5 = 0x2	R6 = 0x0	R7 = 0x3
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFF	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFE
Register file thread id 7:
	R0 = 0x1	R1 = 0xFFFFFFFF	R2 = 0x0	R3 = 0x1	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x2
Blocked MT CPI for this program 2.909091

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0xFFFFFFFD	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0xFFFFFFFF	R1 = 0x3	R2 = 0xFFFFFFFD	R3 = 0x4	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0xFFFFFFFF	R1 = 0x0	R2 = 0x3	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x3	R7 = 0x0
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x4	R6 = 0x0	R7 = 0x0
Register file thread id 4:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0xFFFFFFFF	R7 = 0x1
Register file thread id 5:
	R0 = 0x0	R1 = 0xFFFFFFFE	R2 = 0x4	R3 = 0x0	R4 = 0x0	R5 = 0x2	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0xFFFFFFFF	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFE
Register file thread id 7:
	R0 = 0x1	R1 = 0x3	R2 = 0x0	R3 = 0x1	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x2
Finegrained Multithreading CPI for this program 1.090909

